climada/hazard/trop_cyclone.py

```diff
@@ -334,14 +346,17 @@ def _tc_from_track(self, track, centroids, coastal_idx, model='H08',
                                                              metric=metric)
         reachable_coastal_centr_idx = coastal_idx[reachable_centr_idx]
         npositions = windfields.shape[0]
-        intensity = np.zeros(ncentroids)
-        intensity[reachable_coastal_centr_idx] = np.linalg.norm(windfields, axis=-1)\
-                                                          .max(axis=0)
+
+        intensity = np.linalg.norm(windfields, axis=-1).max(axis=0)
         intensity[intensity < self.intensity_thres] = 0
+        intensity = sparse.csr_matrix(
+            (intensity, reachable_coastal_centr_idx, [0, intensity.size]),
+            shape=(1, ncentroids))
```

Comment:
Instead of first allocating a temporary dense array of size `ncentroids`, we directly write the data to a sparse array. Usually this doesn't make a big difference because `ncentroids` is not that huge and garbage collection is really fast. But I think it's still more elegant since we already have the non-sparse indices (`reachable_coastal_centr_idx`) anyway.

Reply:
If I am not mistaken, the variable `intensity` changes type between line 350 and 352. Having a variable change type during a code is I think a bad habit.